File: sol/solution/0001-0100/0034. Find First and Last Position of Element in Sorted Array/Solution.dart
class Solution {
  List<int> searchRange(List<int> nums, int target) {
    int findBound(bool first) {
      int l = 0, r = nums.length - 1, ans = -1;
      while (l <= r) {
        int mid = (l + r) >> 1;
        if (nums[mid] == target) {
          ans = mid;
          if (first) r = mid - 1;
          else l = mid + 1;
        } else if (nums[mid] < target) l = mid + 1;
        else r = mid - 1;
      }
      return ans;
    }
    return [findBound(true), findBound(false)];
  }
}

File: sol/solution/0001-0100/0034. Find First and Last Position of Element in Sorted Array/Solution.kt
class Solution {
    fun searchRange(nums: IntArray, target: Int): IntArray {
        fun findBound(first: Boolean): Int {
            var l = 0; var r = nums.size - 1; var ans = -1
            while (l <= r) {
                val mid = (l + r) / 2
                if (nums[mid] == target) {
                    ans = mid
                    if (first) r = mid - 1 else l = mid + 1
                } else if (nums[mid] < target) l = mid + 1
                else r = mid - 1
            }
            return ans
        }
        return intArrayOf(findBound(true), findBound(false))
    }
}